app/ui/src/main/java/com/fsck/k9/activity/ChooseFolder.java

```diff
@@ -367,29 +364,171 @@ public void run() {
         }
     };
 
-    private String buildDisplayName(Account account, String serverId, String name) {
-        if (account.getInboxFolder().equals(serverId)) {
-            return getString(R.string.special_mailbox_name_inbox);
-        } else {
-            return name;
+    class FolderListAdapter extends BaseAdapter implements Filterable {
+        private List<FolderInfoHolder> mFolders = new ArrayList<>();
+        private List<FolderInfoHolder> mFilteredFolders = Collections.unmodifiableList(mFolders);
+        private Filter mFilter = new FolderListAdapter.FolderListFilter();
+        private FolderIconProvider folderIconProvider = new FolderIconProvider(getTheme());
+
+        public FolderInfoHolder getItem(long position) {
+            return getItem((int)position);
         }
-    }
 
+        public FolderInfoHolder getItem(int position) {
+            return mFilteredFolders.get(position);
+        }
+
+
+        public long getItemId(int position) {
+            return mFilteredFolders.get(position).folder.getServerId().hashCode() ;
```

Comment:
This is a bug. I created #4180 to fix it (and clean up `FolderListAdapter` a bit). You probably want to wait until #4180 has been merged.